src-self-hosted/translate_c.zig

```diff
@@ -1293,7 +1293,37 @@ fn transBinaryOperator(
     }
 
     const rhs_node = try transExpr(rp, scope, ZigClangBinaryOperator_getRHS(stmt), .used, .r_value);
-    return transCreateNodeInfixOp(rp, scope, lhs_node, op_id, op_token, rhs_node, result_used, true);
+
+    const is_lhs_bool = isBoolRes(lhs_node);
+    const is_rhs_bool = isBoolRes(rhs_node);
+
+    if (!is_lhs_bool and !is_rhs_bool) {
+        return transCreateNodeInfixOp(rp, scope, lhs_node, op_id, op_token, rhs_node, result_used, true);
+    }
+
+    const lhs = if (is_lhs_bool) init: {
+        const cast_node = try transCreateNodeBuiltinFnCall(rp.c, "@boolToInt");
+        try cast_node.params.push(lhs_node);
+        cast_node.rparen_token = try appendToken(rp.c, .RParen, ")");
+        break :init &cast_node.base;
+    } else lhs_node;
+
+    const rhs = if (is_rhs_bool) init: {
+        const cast_node = try transCreateNodeBuiltinFnCall(rp.c, "@boolToInt");
+        try cast_node.params.push(rhs_node);
+        cast_node.rparen_token = try appendToken(rp.c, .RParen, ")");
+        break :init &cast_node.base;
+    } else rhs_node;
+
+    const node = try rp.c.a().create(ast.Node.InfixOp);
+
+    node.* = .{
+        .op_token = op_token,
+        .lhs = lhs,
+        .op = op_id,
+        .rhs = rhs,
+    };
+    return maybeSuppressResult(rp, scope, result_used, &node.base);
```

Comment:
This should use `transCreateNodeInfixOp` since it takes care of grouping the expression.